### DISCUSSION TITLE
Welcome to botan Discussions!

### DISCUSSION THREAD

**Discussion by randombit** [randombit/botan#2859] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by hrantzsch** [comment]
Hi Botan community!

I'm Hannes, a security and crypto engineer from Berlin :wave:  I work at neXenio, where we first started using Botan around five years ago for our secure file sharing solution. Currently we're involved in adding support for TLS 1.3 and adding Kyber.

Thanks @randombit for setting up Discussions, I'm looking forward to building a stronger Botan community with all of you :smiley:

**Comment by reneme** [comment]
Hello Botan people!

I'm René, working on security/crypto related projects at [neXenio](https://www.nexenio.com) in Germany.

As @hrantzsch already pointed out, we're using Botan in our company for quite a while now. A major factor in that decision was that Botan is [endorsed by the German Federal Office for Information Security](https://www.bsi.bund.de/EN/Topics/Cryptography/CryptoLibrary/crypto_library.html) for use in high security applications.

I started contributing to the library with improvements on the X.509 certificate support, integration of TLS trust roots provided by the operating system and also a few build systems improvements (e.g. [Conan-Center's botan recipe](https://github.com/conan-io/conan-center-index/tree/master/recipes/botan)).

Lately, I'm helping to add Kyber as a PQC algorithm (via the [KBLS](https://www.forschung-it-sicherheit-kommunikationssysteme.de/projekte/kbls) project) and work with @hrantzsch on the implementation of TLS 1.3.

Thanks @randombit, for this amazing library and the great experience while contributing to it!

**Comment by omaralam96** [comment]
I am starting using Botan 3.1, and as it stated that Botan 3.x based on C++20  while my project is implemented using c++14.
Would that cause any trouble for me !?

**Reply by reneme** [reply]
Please see my reply here: https://github.com/randombit/botan/discussions/3631#discussioncomment-6478359

**Comment by KaganCanSit** [comment]
Hello Botan Community,

I am Kağan. I have been a professional software engineer for about three years. I am an open source and C++ enthusiast. I am currently working especially on Linux-based server software and I use Botan for content verification (cross-checking).

I like Botan's modern C++20 standard, up-to-date tools, and tightly controlled development environment. Although I still stick to C++11 and older technologies in my business life, I have the opportunity to learn by realizing my own mistakes thanks to project tests and development standards.

In this direction, I want to improve both Botan and my own skills by exchanging information with you and making small contributions.

Thank you @randombit for what you have developed and presented. Also, thanks to @reneme who guided me with her knowledge most of the time during the review phase.

I am always open to your feedback and especially want to support the sustainable development of the project in my free time.

Have a nice day!

**Comment by kre-medisoftware** [comment]
Hi there,
Botan is great so far. I'm missing some examples to sign a JWT.
The best way to work with JWTs is today [JWT by Thalhammer](https://github.com/Thalhammer/jwt-cpp) which needs openssl.
Botan would be an excellent alternative to get rid of openssl.

Thanks

**Reply by randombit** [reply]
Thanks for the kind words. I do agree JWT would be useful and in scope, we'd have to figure out what we're doing for JSON (this is also an issue blocking supporting the ACME protocol). However I note that this JWT library appears to handle multiple JSON libraries via https://github.com/Thalhammer/jwt-cpp/blob/master/docs/traits.md I'd have to look at it further but perhaps we can just adopt that idea.

**Reply by dmazzella** [reply]
> Hi there, Botan is great so far. I'm missing some examples to sign a JWT. The best way to work with JWTs is today [JWT by Thalhammer](https://github.com/Thalhammer/jwt-cpp) which needs openssl. Botan would be an excellent alternative to get rid of openssl.
> 
> Thanks

Hi,

I had the same need and built a header-only C++20 JOSE library backed by Botan (published just a short while ago): [jose](https://github.com/dmazzella/jose)
It covers the full JOSE family — JWT, JWS, JWK, and JWE — with no OpenSSL dependency:

```cpp
#include <jose/jwt/jwt_botan_jsoncons.hpp>

std::error_code ec;
std::string token = jwt::create()
    .set_issuer("my-service")
    .sign(jwt::algorithm::hs256("secret"), ec);

auto decoded = jwt::decode(token, ec);
jwt::verify()
    .with_issuer("my-service")
    .allow_algorithm(jwt::algorithm::hs256("secret"))
    .verify(decoded, ec);
```

Supported algorithms: HS256/384/512, RS256/384/512, PS256/384/512, ES256/384/512.
The API is heavily inspired by jwt-cpp/Thalhammer so the mental model should feel familiar.

The JSON backend is traits-based (C++20 concepts), though for now only [jsoncons](https://github.com/danielaparker/jsoncons) is provided but is easy to implement news for example [nlohmann json](https://github.com/nlohmann/json)

Hope it helps!

**Reply by kre-medisoftware** [reply]
Hi [dmazzella](https://github.com/dmazzella),
thanks for this reply!!! It's worked fine for me. Very smart and easy to use. The need for [jsoncons](https://github.com/danielaparker/jsoncons) is acceptable, maybe [nlohmann](https://github.com/nlohmann/json) - support will follow, which i would prefer.

**Reply by dmazzella** [reply]
> Hi [dmazzella](https://github.com/dmazzella), thanks for this reply!!! It's worked fine for me. Very smart and easy to use. The need for [jsoncons](https://github.com/danielaparker/jsoncons) is acceptable, maybe [nlohmann](https://github.com/nlohmann/json) - support will follow, which i would prefer.

@kre-medisoftware added [nlohmann](https://github.com/nlohmann/json) support
